solr/solrj/src/java/org/apache/solr/common/cloud/ZkNodeProps.java

```diff
@@ -105,6 +111,17 @@ public static ZkNodeProps load(byte[] bytes) {
 
   @Override
   public void write(JSONWriter jsonWriter) {
+    final String baseUrl = (String)propMap.get(ZkStateReader.BASE_URL_PROP);
+    if (baseUrl != null && !baseUrl.startsWith(SCHEME_VAR)) {
+      final int at = baseUrl.indexOf("://");
+      if (at != -1) {
+        final String updatedUrl = SCHEME_VAR + baseUrl.substring(at+3);
+        Map<String,Object> modMap = new LinkedHashMap<>(propMap);
+        modMap.put(ZkStateReader.BASE_URL_PROP, updatedUrl);
+        jsonWriter.write(modMap);
```

Comment:
I think it's clearer here if we only have a single return from the method, so we should fall through or have an else.